### ISSUE TITLE
Library shuffle() method does not shuffle properly

### ISSUE THREAD

**Issue by Ivaldes77** [magefree/mage#5203]
When shuffle method is called from player.shuffleLibrary(game), the shuffle algorithm dosen't take into account a big amount of posibilities.

This is the shuffle function located at mage/Mage/src/main/java/mage/players/Library.java

<pre>
 public void shuffle() {
        UUID[] shuffled = library.toArray(new UUID[0]);
        for (int n = shuffled.length - 1; n > 0; n--) {
          <b>  int r = RandomUtil.nextInt(n); </b>
            UUID temp = shuffled[n];
            shuffled[n] = shuffled[r];
            shuffled[r] = temp;
        }
        library.clear();
        library.addAll(Arrays.asList(shuffled));
    }

</pre>
If the library size is 10 n will take values from 9 to 1. The problem here, is when calling the RandomUtil.nextInt() method with n variable as a parameter.

```
  public static int nextInt(int max) {
        return ThreadLocalRandom.current().nextInt(max);
    }
```
_From: mage/Mage/src/main/java/mage/util/RandomUtil.java_

ThreadLocalRandom's nextInt(bound) takes this upper bound as exclusive as specified in the java docs

> nextInt(int bound)
> Returns a pseudorandom int value between zero (inclusive) and the specified bound (exclusive).

_From https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#nextInt-int-_

This means that when a card is shuffled, it will never stay in the same position as r will never take the same value of n. This only takes  into account ~37% of the total chances that should be possible in a normal shuffling.

To solve this, this line should be changed

`  int r = RandomUtil.nextInt(n); `

for this

`  int r = RandomUtil.nextInt(n+1); `

**Comment by quercitron** [comment]
Hi! Your comment is correct, but as far as I can see this problem was fixed here https://github.com/magefree/mage/commit/d1fd69f685c9d92e14f0e34d8a72f4eae8991a66. What code version did you check?
Thank you.

**Comment by Ivaldes77** [comment]
True, we were checking a previous branch. Thanks for your response and sorry for the mistake.
